aten/src/ATen/cuda/detail/KernelUtils.h

```diff
@@ -28,4 +31,10 @@ inline int GET_BLOCKS(const int N)
   return (N + CUDA_NUM_THREADS - 1) / CUDA_NUM_THREADS;
 }
 
+inline int GET_BLOCKS(const int64_t N) {
+  AT_ASSERTM(N > 0, "CUDA kernel launch blocks must be positive, but got N=", N);
+  constexpr int64_t max_int = std::numeric_limits<int>::max();
+  return GET_BLOCKS(static_cast<int>(std::min(max_int, N)));
+}
+
```

Comment:
How did you decide how to pick the number of blocks?

Reply:
This is the same logic as before, except if `N` doesn't fit in `int` I consider it to be `INT_MAX`. That gives a maximum cuda thread count of `2**31 - CUDA_NUM_THREADS`. This should be more than enough though since `CUDA_KERNEL_LOOP` is looping over the full range even if there are fewer threads than loop iterations.

Reply:
Sounds reasonable, thanks for the explanation